<launch>

  <node pkg="cam_imu_sync" type="sync_test_node" name="sync_test" output="screen">

      <!-- General parameters  -->
      <param name="input_bagfile_name" type="string" value="/home/ke/Dataset/calibration/calibration_2015-10-22-16-59-58.bag" />
      <param name="output_bagfile_name" type="string" value="/home/ke/Dataset/calibration/calibration_2015-10-22-16-59-58_processed.bag" />
      <param name="input_imu_topic" type="string" value="/imu/imu" />
      <param name="input_img_topic" type="string" value="/mv_25001298/image_rect" />
      <param name="output_imu_angular_vel_topic" type="string" value="/sync/imu" />
      <param name="output_cam_angular_vel_topic" type="string" value="/sync/cam" />

      <!-- Calibration parameters  -->
      <param name="ransac_threshold" type="double" value="2" />
      <param name="ransac_confidence" type="double" value="0.8" />
      <param name="fx" type="double" value="365.2417602539062" />
      <param name="fy" type="double" value="367.1694595448353" />
      <param name="cx" type="double" value="426.9591369628906" />
      <param name="cy" type="double" value="234.2825454875092" />
      <param name="skew" type="double" value="0.0" />
      <param name="ic_offset_roll" type="double" value="0.0" />
      <param name="ic_offset_pitch" type="double" value="0.0" />
      <param name="ic_offset_yaw" type="double" value="3.141593" />

      <!-- Detection parameters  -->
      <param name="detection/window_size" type="int" value="5" />
      <param name="detection/suppression" type="int" value="7" />
      <param name="detection/descriptor" type="int" value="5" />
      <param name="detection/quality_level" type="double" value="0.001" />
      <param name="detection/grid_height" type="int" value="6" />
      <param name="detection/grid_width" type="int" value="8" />
      <param name="detection/grid_max_corner_number" type="int" value="3" />

      <!-- Tracking parameters  -->
      <param name="tracking/pyramid_level" type="int" value="4" />
      <param name="tracking/window_size" type="int" value="15" />
      <param name="tracking/klt_max_iterations" type="int" value="15" />
      <param name="tracking/convergence" type="double" value="0.1" />
      <param name="tracking/klt_threshold" type="int" value="4" />
      <param name="tracking/block_size" type="int" value="5" />
  </node>

</launch>

